integration-tests/src/test/java/com/datastax/oss/driver/api/core/metadata/SerializableMetadataIT.java

```diff
@@ -0,0 +1,153 @@
+            "CREATE TYPE IF NOT EXISTS numbers_data (doub int, str text)",
+            "CREATE TABLE IF NOT EXISTS smaller (val int primary key, data numbers_data)",
+            "CREATE INDEX IF NOT EXISTS doub_idx on all (doub)",
+            "CREATE OR REPLACE FUNCTION avgState ( state tuple<int,bigint>, val int ) CALLED ON NULL INPUT RETURNS tuple<int,bigint> LANGUAGE java AS"
+                + " 'if (val !=null) { state.setInt(0, state.getInt(0)+1); state.setLong(1, state.getLong(1)+val.intValue()); } return state;'",
+            "CREATE OR REPLACE FUNCTION avgFinal ( state tuple<int,bigint> ) CALLED ON NULL INPUT RETURNS double LANGUAGE java AS"
+                + " 'double r = 0; if (state.getInt(0) == 0) return null; r = state.getLong(1); r/= state.getInt(0); return Double.valueOf(r);'",
+            "CREATE AGGREGATE IF NOT EXISTS average ( int ) SFUNC avgState STYPE tuple<int,bigint> FINALFUNC avgFinal INITCOND (0,0)",
+            "TRUNCATE all",
+            "INSERT INTO all (val,doub,str) VALUES (1,2,'one')",
+            "INSERT INTO all (val,doub,str) VALUES (2,4,'two')",
+            "INSERT INTO all (val,doub,str) VALUES (3,6,'three')",
+            "CREATE MATERIALIZED VIEW IF NOT EXISTS big AS SELECT * FROM all"
+                + " WHERE val is not null and doub is not null and val > 2"
+                + " PRIMARY KEY (val,doub)");
+
+    CqlSession session = sessionRule.session();
+    statements.forEach(session::execute);
+  }
+
+  private KeyspaceMetadata getAndValidateKsMetadata() {
+
+    Optional<KeyspaceMetadata> ksOption =
+        sessionRule.session().getMetadata().getKeyspace(sessionRule.session().getKeyspace().get());
+    assertTrue(ksOption.isPresent());
+    KeyspaceMetadata ks = ksOption.get();
+    assertTrue(ks instanceof DefaultKeyspaceMetadata);
+    return ks;
+  }
+
+  @Test
+  public void should_contain_serializable_metadata() {
+
+    KeyspaceMetadata ks = getAndValidateKsMetadata();
+
+    /* Validate that the keyspace metadata is fully populated */
+    assertFalse(ks.getUserDefinedTypes().isEmpty());
+    assertFalse(ks.getTables().isEmpty());
+    assertFalse(ks.getViews().isEmpty());
+    assertFalse(ks.getFunctions().isEmpty());
+    assertFalse(ks.getAggregates().isEmpty());
+
+    Optional<TableMetadata> tableOption = ks.getTable("all");
+    assertTrue(tableOption.isPresent());
+    TableMetadata table = tableOption.get();
+    assertTrue(table instanceof DefaultTableMetadata);
+
+    /* Validate that the table metadata is fully populated */
+    assertFalse(table.getPartitionKey().isEmpty());
+    assertFalse(table.getClusteringColumns().isEmpty());
+    assertFalse(table.getColumns().isEmpty());
+    // TODO: What kind of objects can be set for options here?  How do we guarantee they're
+    // serializable?
+    assertFalse(table.getOptions().isEmpty());
+    assertFalse(table.getIndexes().isEmpty());
+
+    SerializationUtils.roundtrip((DefaultKeyspaceMetadata) ks);
```

Comment:
Nit -- we have a built-in utility for that: `SerializationHelper.serializeAndDeserialize`.

It's essentially the same thing, but we wrote it at a time when commons-lang wasn't yet available in the classpath (I'm not even sure how it landed here TBH). Let's use it everywhere for the sake of consistency.

Reply:
Looks like it's Gremlin's fault:

```
[INFO] +- org.apache.tinkerpop:gremlin-core:jar:3.4.5:compile
[INFO] |  +- org.apache.tinkerpop:gremlin-shaded:jar:3.4.5:compile
[INFO] |  +- commons-configuration:commons-configuration:jar:1.10:compile
[INFO] |  |  \- commons-lang:commons-lang:jar:2.6:compile
[INFO] |  +- commons-collections:commons-collections:jar:3.2.2:compile
[INFO] |  +- org.apache.commons:commons-lang3:jar:3.8.1:compile
[INFO] |  +- org.yaml:snakeyaml:jar:1.15:compile
[INFO] |  +- com.carrotsearch:hppc:jar:0.7.1:compile
[INFO] |  +- com.jcabi:jcabi-manifests:jar:1.1:compile
[INFO] |  |  \- com.jcabi:jcabi-log:jar:0.14:compile
[INFO] |  +- com.squareup:javapoet:jar:1.11.1:compile
[INFO] |  +- net.objecthunter:exp4j:jar:0.4.8:compile
[INFO] |  \- org.slf4j:jcl-over-slf4j:jar:1.7.25:compile
```

I love that the same artifact brings in both commons-lang2 (via a transitive dep) and commons-lang3 directly. :rage: